cmd/notary/tuf.go

```diff
@@ -851,23 +851,37 @@ type passwordStore struct {
 	anonymous bool
 }
 
+func getUsername(input chan string){
+	in := bufio.NewReader(os.Stdin)
+	result, err := in.ReadString('\n')
+	if err != nil {
+		logrus.Errorf("error processing username input: %s", err)
+		input <- ""
+	}
+	input <- result
+}
+
 func (ps passwordStore) Basic(u *url.URL) (string, string) {
 	// if it's not a terminal, don't wait on input
 	if ps.anonymous {
 		return "", ""
 	}
 
 	stdin := bufio.NewReader(os.Stdin)
+	input := make(chan string, 1)
 	fmt.Fprintf(os.Stdout, "Enter username: ")
-
-	userIn, err := stdin.ReadBytes('\n')
-	if err != nil {
-		logrus.Errorf("error processing username input: %s", err)
-		return "", ""
+	go getUsername(input)
+	var username string
+	select {
+	case i := <-input:
+		username = strings.TrimSpace(i)
+		if username == "" {
+			return "",""
+		}
+	case <-time.After(30 * time.Second):
+		return "",""
```

Comment:
can we add an error log here?  Something like `timeout when retrieving username input`?